blockchain/client.go

```diff
@@ -215,6 +215,38 @@ func (c *Client) ListenEvents(
 	return g.Wait()
 }
 
+func forwardHeaders(ctx context.Context, from, to chan types.Header) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case header, ok := <-from:
+			if !ok {
+				return context.Canceled
```

Comment:
`return nil` or some custom error looks like a better fit here. Otherwise we receive `Cancelled` somewhere higher level and skip retrying thinking it is our intentional cancellation. What do you think?

Reply:
added custom err